src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -946,38 +966,40 @@ bool StatefulWriter::matched_reader_add(
                 }
                 else
                 {
-                    not_relevant_changes.insert(current_seq);
-                }
-                ++current_seq;
-            }
-
-            ChangeForReader_t changeForReader(*cit);
-
-            if (rp->durability_kind() >= TRANSIENT_LOCAL && this->getAttributes().durabilityKind >= TRANSIENT_LOCAL)
-            {
-                changeForReader.setRelevance(rp->rtps_is_relevant(*cit));
-                if (!rp->rtps_is_relevant(*cit))
-                {
-                    if (rp->is_local_reader())
+                    try
                     {
-                        intraprocess_gap(rp, current_seq);
+                        gap_builder.add(current_seq);
```

Comment:
Not send sequence number minor than history sequence number.

Reply:
current_seq starts on get_seq_num_min()